xdf.h

```diff
@@ -54,7 +55,7 @@ class Xdf
     struct Stream
     {
         //! A 2D vector which stores the time series of a stream. Each row represents a channel.
-        std::vector<std::vector<float> > time_series;
+        std::vector<std::vector<std::variant<int, float, double, int64_t, std::string>>> time_series;
```

Comment:
Not sure if a template would be better here. std::variant here does not seem to be very convenient.

Reply:
If a template is a better solution here, I am totally open to it.  I do 99% of my programming in R, and I've done just enough C++ programming in the past that I can do the basics.  I am not going to pretend that I'm good at it - but as Rcpp is the best method of reading the xdf files into R, and my group does produce streams with strings, I was motivated to get a C++ solution working.  If you have a better idea on how to address strings, feel free to scrap this - this is a "something that works" solution, not what I would call a perfect/elegant solution.  Then again, my C++ is bad enough that I might not recognize the difference.

I did try to use a template to read in the data, after I got this working on my computer (still storing it using std::variant), and I went back to what I had before because I couldn't get it to work.   If you're interested in going that direction, I'm unlikely to have the skills to make that happen very quickly.

Reply:
Thank you very much. I think for all the numeric types, template is probably much easier to do, actually (I provided an example in one of the comments). Strings might be a little tricky, but worth a try and please feel free to let me know if you encounter any unexpected issues.